src/System.Text.Primitives/System/Text/Encoders/Utf8Utility.cs

```diff
@@ -0,0 +1,357 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Buffers.Text
+{
+    /// <summary>
+    /// Contains utility methods for inspecting UTF-8 sequences.
+    /// </summary>
+    /// <remarks>
+    /// The UTF-8 specification can be found in the Unicode Standard, Sec. 3.9.
+    /// http://www.unicode.org/versions/Unicode10.0.0/ch03.pdf
+    /// </remarks>
+    public static class Utf8Utility
+    {
+        private static readonly byte[] _replacementCharAsUtf8 = new byte[] { 0xEF, 0xBF, 0xBD };
+
+        /// <summary>
+        /// The Unicode Replacement Character (U+FFFD) as the three-byte UTF-8 sequence [ EF BF BD ].
+        /// </summary>
+        public static ReadOnlySpan<byte> ReplacementCharacterByteSequence => _replacementCharAsUtf8;
+
+        /// <summary>
+        /// Given the first byte of a sequence, returns the expected number of continuation bytes
+        /// which should follow this byte.
+        /// </summary>
+        /// <remarks>
+        /// Returns 0 if <paramref name="firstByte"/> does not have any expected continuation bytes
+        /// or if <paramref name="firstByte"/> cannot begin a well-formed UTF-8 sequence.
+        /// </remarks>
+        public static int GetExpectedNumberOfContinuationBytes(byte firstByte)
```

Comment:
Do we need this to be public?

Reply:
I don't have a use case for this API right now; will just remove until it's needed.